homeassistant/components/home_connect/services.py

```diff
@@ -0,0 +1,567 @@
+                option_key=option_key,
+                value=value,
+                unit=unit,
+            )
+        else:
+            await client.set_selected_program_option(
+                ha_id,
+                option_key=option_key,
+                value=value,
+                unit=unit,
+            )
+    except HomeConnectError as err:
+        raise HomeAssistantError(
+            translation_domain=DOMAIN,
+            translation_key="set_options_active_program"
+            if active
+            else "set_options_selected_program",
+            translation_placeholders={
+                **get_dict_from_home_connect_error(err),
+                "key": option_key,
+                "value": str(value),
+            },
+        ) from err
+
+
+async def _async_service_command(call: ServiceCall, command_key: CommandKey) -> None:
+    """Execute calls to services executing a command."""
+    client, ha_id = await _get_client_and_ha_id(call.hass, call.data[ATTR_DEVICE_ID])
+
+    async_create_issue(
+        call.hass,
+        DOMAIN,
+        "deprecated_command_actions",
+        breaks_in_ha_version="2025.9.0",
+        is_fixable=True,
+        is_persistent=True,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_command_actions",
+    )
+
+    try:
+        await client.put_command(ha_id, command_key=command_key, value=True)
+    except HomeConnectError as err:
+        raise HomeAssistantError(
+            translation_domain=DOMAIN,
+            translation_key="execute_command",
+            translation_placeholders={
+                **get_dict_from_home_connect_error(err),
+                "command": command_key.value,
+            },
+        ) from err
+
+
+async def async_service_option_active(call: ServiceCall) -> None:
+    """Service for setting an option for an active program."""
+    await _async_service_set_program_options(call, True)
+
+
+async def async_service_option_selected(call: ServiceCall) -> None:
+    """Service for setting an option for a selected program."""
+    await _async_service_set_program_options(call, False)
+
+
+async def async_service_setting(call: ServiceCall) -> None:
+    """Service for changing a setting."""
+    key = call.data[ATTR_KEY]
+    value = call.data[ATTR_VALUE]
+    client, ha_id = await _get_client_and_ha_id(call.hass, call.data[ATTR_DEVICE_ID])
+
+    try:
+        await client.set_setting(ha_id, setting_key=key, value=value)
+    except HomeConnectError as err:
+        raise HomeAssistantError(
+            translation_domain=DOMAIN,
+            translation_key="set_setting",
+            translation_placeholders={
+                **get_dict_from_home_connect_error(err),
+                "key": key,
+                "value": str(value),
+            },
+        ) from err
+
+
+def register_actions(hass: HomeAssistant) -> None:
+    """Register custom actions."""
+
+    async def async_service_pause_program(call: ServiceCall) -> None:
```

Comment:
We could move all the service handlers to the module level if we use `call.hass` consistently.

Reply:
True, I forgot to move those. Thank you!